packages/sonarwhal/src/lib/utils/npm.ts

```diff
@@ -8,27 +6,48 @@ import { spawnSync, SpawnSyncReturns } from 'child_process';
 import { NpmPackage } from '../types';
 import { debug as d } from './debug';
 import * as logger from './logging';
+import { loadJSONFile, findPackageRoot } from './misc';
 
 const debug: debug.IDebugger = d(__filename);
 const npmLoadAsync = promisify(npm.load);
 
-
-const packageExists = () => {
-    const packagePath: string = path.join(process.cwd(), 'package.json');
-
-    return fs.existsSync(packagePath); // eslint-disable-line no-sync
-};
-
 export const installPackages = (packages: Array<string>): boolean => {
-    const global: boolean = !packageExists();
+    /** Whether or not the package should be installed globally. */
+    let global: boolean = false;
+    /** Whether or not the package should be installed as devDependencies. */
+    let isDev: boolean = false;
+    /** Path to `package.json`. */
+    let packagePath: string;
+    /** Current working directory. */
+    const currentWorkingDir = process.cwd();
+    /** Command to install the packages. */
+    let command: string = `npm install ${packages.join(' ')}`;
 
     if (packages.length === 0) {
         return true;
     }
 
-    const command: string = `npm install ${packages.join(' ')}${global ? ' -g' : ''}`;
+    try {
+        packagePath = findPackageRoot(currentWorkingDir);
+
+        global = packagePath !== currentWorkingDir;
+    } catch (error) {
+        // `package.json` is not found.
+        global = true;
+    }
 
     try {
+        if (!global) {
+            const jsonContent = loadJSONFile(packagePath);
+
+            // If `sonarwhal` is a devDependency, then set all packages as devDependencies.
+            isDev = jsonContent && jsonContent.devDependencies.hasOwnProperty('sonarwhal');
```

Comment:
I think `devDependencies` is not mandatory in the file `package.json`. If so, you need to check if `devDependencies` exists before call to `hasOwnProperty`.